### PR TITLE
Fix string type in bundle loader

### PR DESCRIPTION
**Pull request by buchi**
With this fix all items injected into the transmogrifier pipeline contain utf8 encoded byte strings instead of unicode strings. This fixes an issue with unicode strings in local roles (#2927).

It may introduce new issues if something in the ogg bundle pipeline relies on unicode strings. But in general items in the transmogrifier pipeline should always contain byte strings. Various sections will break with unicode strings.

/cc @lukasgraf

### CODE REVIEW

File: opengever/bundle/loader.py [hunk 1]
```diff
@@ -244,3 +246,17 @@ def __call__(self):
...
+
+
+def unicode2bytes(data):
+    """Converts unicode strings to byte strings."""
+    if isinstance(data, unicode):
+        return data.encode('utf8')
+    elif isinstance(data, str):
+        return data
+    elif isinstance(data, Mapping):
+        return dict(map(unicode2bytes, data.iteritems()))
```
Comment (deiferni) [inline opengever/bundle/loader.py]: should we also attempt preserve the mapping-type here?

File: opengever/bundle/loader.py [hunk 2]
```diff
@@ -244,3 +246,17 @@ def __call__(self):
             log.info('No ingestion settings found at %s' % settings_path)
             return {}
         return settings
+
+
+def unicode2bytes(data):
```
Comment (deiferni) [inline opengever/bundle/loader.py]: could you provide some unit-tests for this helper function?

Comment (buchi) [inline opengever/bundle/loader.py]: I've added some unit tests, though this is rather trivial code.

File: opengever/bundle/loader.py (hunk 1 above)
Comment (buchi) [inline opengever/bundle/loader.py]: done.

File: opengever/bundle/loader.py (hunk 2 above)
Comment (deiferni) [inline opengever/bundle/loader.py]: 😄 💛

**Comment by buchi** [comment]
The DX schema updater handles byte strings: https://github.com/collective/transmogrify.dexterity/blob/master/transmogrify/dexterity/converters.py#L449

As already said above, it may introduce new issues. But relying on unicode strings in the transmogrifier pipeline is wrong.

**Pull request merged**